File: 2021 January Bronze/udderedbutnotherd.py
def cycles(ab, w):
    count = 0
    i = 0
    while i < len(w):
        for j in range(0, len(ab)):
            if w[i] == ab[j]:
                #print(f'match at i:{i} for {w[i]} and j:{j} for {ab[j]}')
                i+=1
                if(i == len(w)):
                    break
        #print(f'count increase at {i} loop')
        count+=1
    return count

cowphabet = list(str(input()))
word = input()

'''
abcdefghijklmnopqrstvwxyz
mood
'''
print(cycles(cowphabet, word))

File: 2021 January Bronze/udderedbutnotherd.java
import java.util.*;

public class udderedbutnotherd{
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String cowphabet = in.nextLine();
        String word = in.nextLine();
        
        in.close();

        System.out.println(cycles(cowphabet, word));

    }

    public static int cycles(String ab, String w){
        int count = 0;
        int i = 0;
        while(i < w.length()){
            for(int j = 0; j < ab.length(); j++){
                if (w.charAt(i) == ab.charAt(j)){
                    i++;
                    if(i == w.length()) 
                        break;
                }
            }
            count++;
        }
        return count;
        }
}